### PR TITLE
Add super user validation

### PR DESCRIPTION
**Pull request by rockambole86**
### Background

Doppler needs to be able to get Invoice information for any client.

### Changes

Based on the https://github.com/FromDoppler/doppler-custom-domain/pull/13 work, the same logic was applied here

### CODE REVIEW

**Comment by VicThorX** [comment]
💯

File: Billing.API.Test/InvoiceApiTest.cs [hunk 1]
```diff
@@ -42,34 +42,36 @@ public async Task GetInvoices_WhenInvalidPath_ReturnsNotFound(string url)
         }
 
         [Theory]
-        [AutoData]
-        public async Task GetInvoices_WhenValidTokenProd_ReturnsInternalServerError(string host, string expectedUserName, string expectedPassword)
+        [InlineData(HttpStatusCode.OK, "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYmYiOjE1OTc3NjQ1MjIsImV4cCI6MTU5Nzc2NDUzMiwiaWF0IjoxNTk3NzY0NTIyLCJpc1NVIjp0cnVlfQ.j1qzmKcnpCCBoXAtK9QuzCcnkIedK_kpwlrQ315VX_bwuxNxDBeEgKCOcjACUaNnf92bStGVYxXusSlnCgWApjlFG4TRgcTNsBC_87ZMuTgjP92Ou_IHi5UVDkiIyeQ3S_-XpYGFksgzI6LhSXu2T4LZLlYUHzr6GN68QWvw19m1yw6LdrNklO5qpwARR4WEJVK-0dw2-t4V9jK2kR8zFkTYtDUFPEQaRXFBpaPWAdI1p_Dk_QDkeBbmN_vTNkF7JwmqXRRAaz5fiMmcgzFmayJFbM0Y9LUeaAYFSZytIiYZuNitVixWZEcXT_jwtfHpyDwZKY1-HlyMmUJJuVsf2A")]
```
Comment (andresmoschini) [inline Billing.API.Test/InvoiceApiTest.cs:45]: image

Comment (rockambole86) [reply Billing.API.Test/InvoiceApiTest.cs:45]: lols, I triple checked that the token were already expired

**Comment by andresmoschini** [comment]
🚀

**Pull request merged**